### DISCUSSION TITLE
How do I break a source .md file into multiple .html files?

### DISCUSSION THREAD

**Discussion by ptram** [quarto-dev/quarto-cli#5820] [Q&A]
### Description

Hi,

I'm trying to output a Scrivener project into a Quarto website. I've not been able, as of now, to make Quarto break the source .md file, generated by Scrivener, into separate .html files in a new web site.

What I did is this:

1. In Scrivener, I programmed the paragraph styles, so that each level is preceded by the `{{< pagebreak >}}` directive.
2. When generated (compiled), the .md file was processed by Quarto as a website.

The result is a single .html page from the original .md file, together with the standard pages generated by Quarto (about, index).

Is there a different way to force Quarto to break a source single file into multiple .html pages?

Paolo

**Comment by mcanouil** [comment]
Make multiple source documents, you will get multiple outputs.
You can do that inside a project, _i.e._, using `_quarto.yml`.

Also, could tou share a repository or something?

Finally, did you look at <https://forum.literatureandlatte.com/t/scrivener-quarto-a-technical-academic-publishing-workflow/129769>?

**Reply by ptram** [reply]
Mcanouil, I've read everthing releated to Quarto in the Scrivener forum, but I've not been able to find an answer to this issue.

You say that I should start from multiple source files. Unfortunately, Scrivener can only generate a single .md file from all its separate snippets. Sections are then marked with the standard syntax inside the monolithic .md file, but it expect the formatter to split it.

You say that I could split the .md file using `_quarto.yml`. Can you point me to some instructions? I know this file contains the formatting instructions, but I can't find, in the documentation, a command instructing Quarto to split the source file.

Please find attached my project folder. Maybe you can give it a look, and see if you notice something totally wrong?

Thank you!
Paolo

[mmd-from-scriv.zip](https://github.com/quarto-dev/quarto-cli/files/11655299/mmd-from-scriv.zip)

**Comment by ptram** [comment]
By further investigation, it seems that Quarto can't break a source .md file into several .html files. Pagebreak commands are not recognized as commands to split the file into separate pages.

This is a problem when the source .md file is generated by Scrivener, that can only generate a monolithic file when compiling a project. The only workaround I can imagine is to compile individual sections, and then let Quarto assemble a website or book from them.

This would be a big task the first time. I guess it would be much easier with subsequent updates, when one has only to compile the modified/updated sections from Scrivener. The newly compiled .md files would replace the older ones, and put together with the ones that have not changed.

It would obviously be great if Quarto could understand pagebreak commands as instructions to split the source document into separate .html pages. I'm even surprised it doesn't to it already.

Paolo

**Reply by mcanouil** [reply]
You could setup a pre-render script to split the markdown file into any number of files.

Note that an HTML page, is not equivalent to a PDF/Word page.
HTML page is actually an HTML document.
This is why "page breaks" has no meaning in HTML (it has some meaning in Paged.js).

**Comment by iandol** [comment]
Pandoc already does do this for EPUB files, and the `chunkedhtml` mechanisms could be modified upstream to support this (i.e. breaking a input AST to several output files of types other than HTML). I suspect Pandoc is the best place for this feature to be added, which then could be adopted by quarto so that you could more flexibly generate website/book projects from either one or many documents...
